fe/fe-core/src/main/java/com/starrocks/statistic/StatisticAutoCollector.java

```diff
@@ -69,101 +55,139 @@ protected void runAfterCatalogReady() {
-        analyzeJob.setWorkTime(LocalDateTime.MIN);
-
+        AnalyzeJob analyzeJob = new AnalyzeJob(AnalyzeJob.DEFAULT_ALL_ID, AnalyzeJob.DEFAULT_ALL_ID, Collections.emptyList(),
+                AnalyzeType.SAMPLE, ScheduleType.SCHEDULE, Maps.newHashMap(), ScheduleStatus.PENDING, LocalDateTime.MIN);
         GlobalStateMgr.getCurrentAnalyzeMgr().addAnalyzeJob(analyzeJob);
     }
 
-    private void collectStatistics(List<TableCollectJob> allJobs) {
-        // AnalyzeJob-List<TableCollectJob> for update AnalyzeJob status
-        Map<AnalyzeJob, List<TableCollectJob>> analyzeJobMap = Maps.newHashMap();
+    private void updateHealthy() {
+        Map<Long, AnalyzeMeta> analyzeMeta = GlobalStateMgr.getCurrentAnalyzeMgr().getAnalyzeMetaMap();
+        for (Map.Entry<Long, AnalyzeMeta> meta : analyzeMeta.entrySet()) {
+            Database database = GlobalStateMgr.getCurrentState().getDb(meta.getValue().getDbId());
+            OlapTable table = (OlapTable) database.getTable(meta.getValue().getTableId());
+
+            long totalLoadRows = meta.getValue().getUpdateRows();
 
-        for (TableCollectJob tjb : allJobs) {
-            if (!analyzeJobMap.containsKey(tjb.job)) {
-                analyzeJobMap.put(tjb.job, Lists.newArrayList(tjb));
+            long minRowCount = Long.MAX_VALUE;
+            for (Partition partition : table.getPartitions()) {
+                if (partition.getRowCount() < minRowCount) {
+                    minRowCount = partition.getRowCount();
+                }
+            }
+
+            /*
+             * health = totalLoadRows / totalRowCount.
+             * The ratio of the number of modified lines to the total number of lines.
+             * Because we cannot obtain complete table-level information, we use the row count of
+             * the partition with the smallest row count as totalRowCount.
+             * It can be understood that we assume an extreme case where all imported and modified lines
+             * are concentrated in only one partition
+             */
+            double healthy;
+            if (minRowCount == 0) {
+                healthy = 1;
+            } else if (totalLoadRows > minRowCount) {
+                healthy = 0;
             } else {
-                analyzeJobMap.get(tjb.job).add(tjb);
+                healthy = 1 - (double) totalLoadRows / (double) minRowCount;
             }
+
+            meta.getValue().setHealthy(healthy);
+            GlobalStateMgr.getCurrentAnalyzeMgr().addAnalyzeMeta(meta.getValue());
         }
+    }
 
-        for (Map.Entry<AnalyzeJob, List<TableCollectJob>> entry : analyzeJobMap.entrySet()) {
-            AnalyzeJob analyzeJob = entry.getKey();
-            if (analyzeJob.getStatus() == ScheduleStatus.FINISH) {
-                continue;
-            }
+    private List<TableCollectJob> generateFullAnalyzeJobs() {
+        Map<Long, TableCollectJob> allTableJobMap = Maps.newHashMap();
```

Comment:
It looks like not deal the  AnalyzeJobs in GlobalStateMgr.getCurrentAnalyzeMgr(),  how we collect stats which user use create analyze job stmt to create analyze job.

Reply:
If the user enables full collection, the automatic collection task will no longer run. If the user closes the full collection, use the job generated by the automatic collection task to collect